src/librbd/Journal.cc

```diff
@@ -1349,6 +1350,10 @@ void Journal<I>::handle_replay_process_safe(ReplayEntry replay_entry, int r) {
   CephContext *cct = m_image_ctx.cct;
 
   m_lock.Lock();
+  assert(m_state == STATE_REPLAYING ||
+         m_state == STATE_FLUSHING_RESTART ||
+         m_state == STATE_FLUSHING_REPLAY);
```

Comment:
I looked at the core and it failed due to an existing race condition on the transition between REPLAYING -> READY.  I'll open a tracker ticket.